src/implementations/truncation.jl

```diff
@@ -28,82 +28,89 @@ end
 
 # findtruncated
 # -------------
+# Generic fallback
+function findtruncated_svd(values::AbstractVector, strategy::TruncationStrategy)
+    return findtruncated(values, strategy)
+end
+
 # specific implementations for finding truncated values
 findtruncated(values::AbstractVector, ::NoTruncation) = Colon()
 
-function findtruncated(values::AbstractVector, strategy::TruncationKeepSorted)
+function findtruncated(values::AbstractVector, strategy::TruncationByOrder)
     howmany = min(strategy.howmany, length(values))
-    return partialsortperm(values, 1:howmany; by=strategy.by, rev=strategy.rev)
+    return partialsortperm(values, 1:howmany; strategy.by, strategy.rev)
 end
-function findtruncated_sorted(values::AbstractVector, strategy::TruncationKeepSorted)
+function findtruncated_svd(values::AbstractVector, strategy::TruncationByOrder)
+    strategy.by === abs || return findtruncated(values, strategy)
     howmany = min(strategy.howmany, length(values))
-    return 1:howmany
+    return strategy.rev ? (1:howmany) : ((length(values) - howmany + 1):length(values))
 end
 
-# TODO: consider if worth using that values are sorted when filter is `<` or `>`.
-function findtruncated(values::AbstractVector, strategy::TruncationKeepFiltered)
-    ind = findall(strategy.filter, values)
-    return ind
+function findtruncated(values::AbstractVector, strategy::TruncationByFilter)
+    return strategy.filter.(values)::AbstractVector{Bool}
```

Comment:
Is it really worth returning a bool vector here, instead of just indices? Is `findall(strategy.filter, values)` so much worse? Or is that for GPU compatibility?